stream_alert/alert_processor/outputs.py

```diff
@@ -803,3 +805,272 @@ def dispatch(self, **kwargs):
+        headers = {'Content-Type': 'application/json', 'cookie' : self.auth_cookie}
+        params = {'jql': jql,
+                  'maxResults': max_results,
+                  'validateQuery': validate_query,
+                  'fields': fields}
+        issues = []
+
+        resp = self._get_request(search_url, params=params, headers=headers, verify=False)
+
+        try:
+            issues = resp.json()['issues']
+        except (ValueError, KeyError):
+            LOGGER.error('An error occurred while searching Jira [%s] : [%s]',
+                         resp.status_code,
+                         resp.content)
+
+        return issues
+
+    def _create_comment(self, issue_id, comment):
+        """Add a comment to an existing issue
+
+        Args:
+            issue_id (str): The existing issue ID or key
+            comment (str): The body of the comment
+
+        Returns:
+            int: ID of the created comment or False if unsuccessful
+        """
+
+        comment_url = os.path.join(self.base_url, self.COMMENT.format(issue_id))
+        comment_body = {'body' : comment}
+        headers = {'Content-Type': 'application/json', 'cookie' : self.auth_cookie}
+        resp = self._post_request(comment_url, data=comment_body, headers=headers, verify=False)
+        jira_id = False
+
+        try:
+            jira_id = resp.json()['id']
+        except (ValueError, KeyError):
+            LOGGER.error('An error occurred while creating Jira comment [%s] : [%s]',
+                         resp.status_code,
+                         resp.content)
+        return jira_id
+
+
+    def _get_comments(self, issue_id):
+        """Get all comments for an existing Jira issue
+
+        Args:
+            issue_id (str): The existing issue ID or key
+
+        Returns:
+            list: List of comments associated with a Jira issue
+        """
+        comment_url = os.path.join(self.base_url, self.COMMENT.format(issue_id))
+        headers = {'Content-Type': 'application/json', 'cookie' : self.auth_cookie}
+        resp = self._get_request(comment_url, headers=headers, verify=False)
+        comments = []
+
+        try:
+            comments = resp.json()['comments']
+        except (ValueError, KeyError):
+            LOGGER.error('An error occurred while getting Jira comments [%s] : [%s]',
+                         resp.status_code,
+                         resp.content)
+        return comments
+
+    def _get_existing_issue(self, issue_summary, project_key):
+        """Find an existing Jira issue based on the issue summary
+
+        Args:
+            issue_summary (str): The Jira issue summary
+            project_key (str): The Jira project to search
+
+        Returns:
+            int: ID of the found issue or False if existing issue does not exist
+        """
+        jql = 'summary ~ "{}" and project="{}"'.format(issue_summary, project_key)
+        resp = self._search_jira(jql, fields=['id', 'summary'], max_results=1)
```

Comment:
Though this is technically not an issue, I'd suggest calling the `_search_jira` function with the named args in the same order as their declared in the function declaration:  max_results --> validate_query --> fields (or re-order the params in the function declaration if you this order better :) )